samples/excel/85-preview-apis/chart-data-source.yaml

```diff
@@ -0,0 +1,129 @@
+            console.log("Qtr1 data source string: " + dataSourceString.value);
+            console.log("Qtr1 data source type: " + dataSourceType.value);
+          });
+        }
+
+        async function setup() {
+          await Excel.run(async (context) => {
+            // Create a new worksheet called "Sample" and activate it.
+            context.workbook.worksheets.getItemOrNullObject("Sample").delete();
+            const sheet = context.workbook.worksheets.add("Sample");
+            
+            // Create an a table named SalesTable on the Sample worksheet.
+            let expensesTable = sheet.tables.add("A1:E1", true);
+            expensesTable.name = "SalesTable";
+
+            expensesTable.getHeaderRowRange().values = [["Product", "Qtr1", "Qtr2", "Qtr3", "Qtr4"]];        
+            expensesTable.rows.add(null, [
+              ["Frames", 5000, 7000, 6544, 4377],
+              ["Saddles", 400, 323, 276, 651],
+              ["Brake levers", 12000, 8766, 8456, 9812],
+              ["Chains", 1550, 1088, 692, 853],
+              ["Mirrors", 225, 600, 923, 544],
+              ["Spokes", 6005, 7634, 4589, 8765]
+            ]);
+            
+            sheet.getUsedRange().format.autofitColumns();
+            sheet.getUsedRange().format.autofitRows();
+            sheet.activate();
+
+            // Create a line chart based on data from Qtr1 from SalesTable.
+            let dataRange = sheet.getRange("A1:B7");
+            let chart = sheet.charts.add("Line", dataRange, "Auto");
+
+            // Position and style the chart.
+            chart.setPosition("A15", "E30");
+            chart.legend.position = "Right";
+            chart.legend.format.fill.setSolidColor("white");
+            chart.dataLabels.format.font.size = 15;
+            chart.dataLabels.format.font.color = "black";
+
+            await context.sync();
+          });
+        }
+
+        /** Default helper for invoking an action and handling errors. */
+        async function tryCatch(callback) {
+          try {
+            await callback();
+          } catch (error) {
+            // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
+            console.error(error);
+          }
+        }
+    language: typescript
+template:
+    content: |-
+        <section class="ms-font-m">
+          <p>This sample shows how to get information about the data source of a chart series.</p>
+        </section>
+
+        <section class="setup ms-font-m">
+          <h3>Set up</h3>
+          <p>Add a product table and line chart to a Sample worksheet.</p>
+          <button id="setup" class="ms-Button">
+            <span class="ms-Button-label">Add sample data</span>
+          </button>
+        </section>
+
+        <section class="samples ms-font-m">
+          <h3>Try it out</h3>
+          <p>Log information to the console about the data source of the chart series <strong>Qtr1</strong>.</p>
```

Comment:
Instead of this label, could we just have the button say "Log data source of X"?

Reply:
Thank you for the review! Great feedback. I've incorporated all of your changes except this one about the button text. I think the description above the button is helpful and I want to keep it. 

The chart now displays 6 series, and the button text would be "Log data source of Frames". Since "Frames" appears in two places on the worksheet, I want to clarify that this is logging source info for the chart series Frames, not the table Frames, which would make the button text "Log data source of chart series Frames" — and that's too long for a button.